op-node/rollup/types.go

```diff
@@ -306,6 +323,31 @@ func (cfg *Config) Check() error {
 	return nil
 }
 
+// validatePlasmaConfig checks the two approaches to configuring plasma mode.
+// If the legacy values are set, they are copied to the new location. If both are set, they are check for consistency.
+func validatePlasmaConfig(cfg *Config) error {
+	if cfg.LegacyUsePlasma && cfg.PlasmaConfig == nil {
+		// copy from top level to plasma config
+		cfg.PlasmaConfig = &PlasmaConfig{
+			DAChallengeAddress: cfg.LegacyDAChallengeAddress,
+			DAChallengeWindow:  cfg.LegacyDAChallengeWindow,
+			DAResolveWindow:    cfg.LegacyDAResolveWindow,
+		}
```

Comment:
Should we nil out the legacy values so that if this config is written out again they will then be omitted? Not sure it matters at this point in the code though.

Reply:
I don't think we go through a read/write cycle. The two writes are from the genesis creation tool  (values from deploy config) and the network config dump tool (network config dump requires the flag)